database/migration/src/main/resources/metadata_changesets/partition_metadata_table.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog objectQuotingStrategy="QUOTE_ALL_OBJECTS"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <changeSet id="update_indexes_metadata_table" author="cahrens" dbms="mysql,postgresql,hsqldb,mariadb">
+        <comment>Updates indexes on the METADATA_ENTRY table to account for new primary key</comment>
+        <createIndex indexName="IX_METADATA_ENTRY_MJI" tableName="METADATA_ENTRY">
+            <column name="METADATA_JOURNAL_ID"/>
+        </createIndex>
+        <dropIndex tableName="METADATA_ENTRY" indexName="METADATA_WORKFLOW_IDX"/>
+    </changeSet>
+    <changeSet id="compress_and_partition_metadata_table" author="cahrens" dbms="mysql">
+        <!-- Do not run this migration in Terra, where we will have already done the migration/table swap -->
+        <preConditions onFail="MARK_RAN">
+            <!-- If we have already run the migration, the primary key will be composite (count 2) -->
+            <sqlCheck expectedResult="1">
+                SELECT COUNT(COLUMN_NAME)
+                FROM information_schema.KEY_COLUMN_USAGE
+                WHERE TABLE_NAME = 'METADATA_ENTRY'
+                AND CONSTRAINT_NAME = 'PRIMARY'
+            </sqlCheck>
+        </preConditions>
+        <comment>Partition and compress the METADATA_ENTRY table (mysqldb only)</comment>
+        <sql>
+            ALTER TABLE METADATA_ENTRY ROW_FORMAT=COMPRESSED KEY_BLOCK_SIZE=8,
+            DROP PRIMARY KEY,
+            ADD CONSTRAINT PK_METADATA_ENTRY PRIMARY KEY(WORKFLOW_EXECUTION_UUID, METADATA_TIMESTAMP, METADATA_JOURNAL_ID)
```

Comment:
Note that I'm adding a constraint name to the primary key. This is for consistency with postgresql, where we need to have a constraint name (and one of our best practices is to always have a constraint name).